src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/BedEntity.kt

```diff
@@ -1,19 +1,91 @@
 package uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity
 
 import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
 import org.springframework.stereotype.Repository
 import java.util.UUID
+import javax.persistence.ColumnResult
+import javax.persistence.ConstructorResult
 import javax.persistence.Entity
 import javax.persistence.Id
 import javax.persistence.JoinColumn
 import javax.persistence.ManyToOne
+import javax.persistence.NamedNativeQuery
+import javax.persistence.SqlResultSetMapping
 import javax.persistence.Table
 
 @Repository
 interface BedRepository : JpaRepository<BedEntity, UUID> {
   fun findByCode(bedCode: String): BedEntity?
+
+  @Query(nativeQuery = true)
+  fun findAllBedsForPremises(premisesId: UUID): List<DomainBedSummary>
 }
 
+@NamedNativeQuery(
+  name = "BedEntity.findAllBedsForPremises",
+  query =
+  """
+    select cast(b.id as text) as id,
+      cast(b.name as text) as name,
+      cast(r.name as text) as roomName,
+      cast(
+        (
+          case 
+          when(
+            select count(booking.id)
+            from bookings booking
+            where booking.bed_id = b.id
+                  and booking.arrival_date <= CURRENT_DATE
+                  and booking.departure_date >= CURRENT_DATE
+          ) > 0 
+          then 
+            true 
+          else 
+            false
+          end
+        ) as boolean
+      ) as bedBooked,
+      cast(
+        (
+          case 
+          when(
+            select count(lost_bed.id)
+            from lost_beds lost_bed
+            where lost_bed.bed_id = b.id
+                  and lost_bed.start_date <= CURRENT_DATE
+                  and lost_bed.end_date >= CURRENT_DATE
+          ) > 0 
+          then 
+            true 
+          else 
+            false
+          end
+        ) as boolean
+      ) as bedOutOfService
```

Comment:
I don't think the `case`/`when` or the `cast` are necessary for these subqueries:

```suggestion
      (
        select count(booking.id)
        from bookings booking
        where booking.bed_id = b.id
          and booking.arrival_date <= CURRENT_DATE
          and booking.departure_date >= CURRENT_DATE
      ) > 0 as bedBooked,
      (
        select count(lost_bed.id)
        from lost_beds lost_bed
        where lost_bed.bed_id = b.id
          and lost_bed.start_date <= CURRENT_DATE
          and lost_bed.end_date >= CURRENT_DATE
      ) > 0 as bedOutOfService
```

Reply:
Ah, of course! That's much nicer 👍